helpers/utils.js

```diff
@@ -1119,66 +1147,76 @@ module.exports = {
+   * @param {Number} koef How to understand we achieve clean price
+   * @return {Number} Clean price
    */
-  getCleanPrice(items, type, liquidity, smartPrice, koef = 0.02) {
+  getCleanPrice(items, type, liquidity, smartPrice, koef = 3) {
 
     try {
 
       let cleanPrice;
+
       let a = 0; let t = 0; let c = 0; let c_t = 0;
       let d = 0; let d2 = 0; let ct_d2 = 0;
-      const enough_ct_d2 = koef;
       const table = [];
 
+      // Each iteration el.price moves towards to Smart price
+
       for (let i = 0; i < items.length; i++) {
 
         const el = items[i];
+
         if (type === 'asks') {
-          if (items[i].price > smartPrice) break;
+          if (el.price > smartPrice) break;
           a = el.amount;
           t = liquidity['percent50'].amountAsks;
         } else {
-          if (items[i].price < smartPrice) break;
+          if (el.price < smartPrice) break;
           a = el.amount * el.price;
           t = liquidity['percent50'].amountBidsQuote;
         }
 
         d = this.numbersDifferencePercent(el.price, smartPrice) / 100;
-        d2 = d * d;
+        d2 = d * d; // Decreases every iteration. For order with smartPrice (last iteration) it equals 0.
         c += a;
-        c_t = c / t;
-        ct_d2 = c_t / d2;
+        c_t = c / t; // Grows each iteration
+        ct_d2 = c_t / d2; // Grows each iteration. For order with smartPrice (last iteration) it equals Infinity.
 
-        if (!cleanPrice && (ct_d2 > enough_ct_d2 || ct_d2 === Infinity)) {
-          cleanPrice = el.price;
+        if (ct_d2 < koef && items[i + 1]) { // While ct_d2 is less than Koef, consider an order as a cheater price
+          cleanPrice = items[i + 1].price;
         }
 
         // This table is only for logging
-        if (!cleanPrice) {
-          table.push({
-            items: items.length,
-            total: +t.toFixed(2),
-            price: el.price.toFixed(8),
-            d: +d.toFixed(2),
-            d2: +d2.toFixed(4),
-            a: a.toFixed(8),
-            c: +c.toFixed(8),
-            c_t: +c_t.toFixed(5),
-            ct_d2: +ct_d2.toFixed(5),
-          });
-        }
+        table.push({
+          items: items.length,
+          total: +t.toFixed(2),
+          price: el.price.toFixed(8),
+          d: +d.toFixed(2),
+          d2: +d2.toFixed(4),
+          a: a.toFixed(8),
+          c: +c.toFixed(8),
+          c_t: +c_t.toFixed(5),
+          ct_d2: +ct_d2.toFixed(5),
+          isCheater: ct_d2 < koef,
+        });
+
+      } // For i
 
+      if (!cleanPrice) { // Set Clean price as the best bid/ask
+        cleanPrice = items?.[0].price;
       }
 
       // See this table to understand the magic
       // console.table(table);
-      // console.log(`cleanPrice for ${type} and ${smartPrice.toFixed(8)} smart price is ${cleanPrice.toFixed(8)}\n`);
+      // console.log(`Clean price is ${cleanPrice.toFixed(8)} for ${type} when Smart price = ${smartPrice.toFixed(8)} and Koef = ${koef}.\n`);
```

Comment:
should we add [ `debug`](npmjs.com/package/debug) for such logs?

Reply:
It will be helpful.